File: server/router/model_transfer/model_transfer.go
package model_transfer

import (
	"github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type ModelTransferRouter struct{}

func (s *ModelTransferRouter) InitTokenRouter(Router *gin.RouterGroup) {
	tokenRouter := Router.Group("model-transfer/token").Use(middleware.OperationRecord())
	tokenApi := v1.ApiGroupApp.ModelTransferApiGroup.TokenApi
	{
		tokenRouter.POST("create", tokenApi.Create)   // 创建Token
		tokenRouter.POST("update", tokenApi.Update)   // 更新Token
		tokenRouter.POST("delete", tokenApi.Delete)   // 删除Token
		tokenRouter.POST("list", tokenApi.List)       // Token列表
		tokenRouter.POST("detail", tokenApi.Detail)   // Token详情
		tokenRouter.POST("regenerate", tokenApi.Regenerate) // 重新生成Token
	}
}

func (s *ModelTransferRouter) InitReportRouter(Router *gin.RouterGroup) {
	reportRouter := Router.Group("model-transfer/report").Use(middleware.OperationRecord())
	reportApi := v1.ApiGroupApp.ModelTransferApiGroup.ReportApi
	{
		reportRouter.POST("daily", reportApi.DailyReport)         // 日报表
		reportRouter.POST("token-usage", reportApi.TokenUsage)    // Token使用详情
		reportRouter.POST("summary", reportApi.SummaryReport)     // 汇总报表
	}
}

File: web/src/api/modelTransfer.js
import service from '@/utils/request'

// ==================== Token管理 ====================

// 创建Token
export const createToken = (data) => {
  return service({
    url: '/model-transfer/token/create',
    method: 'post',
    data
  })
}

// 更新Token
export const updateToken = (data) => {
  return service({
    url: '/model-transfer/token/update',
    method: 'post',
    data
  })
}

// 删除Token
export const deleteToken = (data) => {
  return service({
    url: '/model-transfer/token/delete',
    method: 'post',
    data
  })
}

// Token列表
export const getTokenList = (data) => {
  return service({
    url: '/model-transfer/token/list',
    method: 'post',
    data
  })
}

// Token详情
export const getTokenDetail = (data) => {
  return service({
    url: '/model-transfer/token/detail',
    method: 'post',
    data
  })
}

// 重新生成Token
export const regenerateToken = (data) => {
  return service({
    url: '/model-transfer/token/regenerate',
    method: 'post',
    data
  })
}

// ==================== 报表查询 ====================

// 日报查询
export const getDailyReport = (data) => {
  return service({
    url: '/model-transfer/report/daily',
    method: 'post',
    data
  })
}

// Token使用详情
export const getTokenUsage = (data) => {
  return service({
    url: '/model-transfer/report/token-usage',
    method: 'post',
    data
  })
}

// 汇总报表
export const getSummaryReport = (data) => {
  return service({
    url: '/model-transfer/report/summary',
    method: 'post',
    data
  })
}
